### PR TITLE
python3Packages.diagrams: fix build

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Fix build by adjusting the [`graphviz`](https://github.com/xflr6/graphviz/blob/master/CHANGES.txt) release.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 134734` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.diagrams</li>
    <li>python39Packages.diagrams</li>
  </ul>
</details>

File: pkgs/development/python-modules/diagrams/default.nix [hunk 1]
```diff
@@ -27,15 +27,19 @@ buildPythonPackage rec {
...
   '';
 
   preConfigure = ''
     patchShebangs autogen.sh
     ./autogen.sh
   '';
 
-  patches = [ ./build_poetry.patch ];
+  patches = [
+    # The build-system section is missing
```
Comment (addict3d) [inline pkgs/development/python-modules/diagrams/default.nix:40]: Thank you for noting why this patch is here!

**Review by addict3d — APPROVED** [review]
looks good. 👍

**Comment by addict3d** [comment]
After further consideration, should we be more defensive here, and just bump the spec to `<0.18.0` rather than any?

Upstream is potentially doing that. [see here](https://github.com/mingrammer/diagrams/pull/561/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R14).

File: pkgs/development/python-modules/diagrams/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/diagrams/default.nix:40]: For humans it's clear but `nixpkgs-hammering` could complain :wink:

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134734` at 1180e88a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.diagrams</li>
    <li>python39Packages.diagrams</li>
  </ul>
</details>

**Pull request merged**